_posts/2020-09-06-how-the-system-tests-works.md

```diff
@@ -0,0 +1,493 @@
+                            .addNewEnv()
+                                .withName("BOOTSTRAP_SERVERS")
+                                .withValue(bootstrapServer)
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("TOPIC")
+                                .withValue(topicName)
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("DELAY_MS")
+                                .withValue("1000")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("LOG_LEVEL")
+                                .withValue("DEBUG")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("MESSAGE_COUNT")
+                                .withValue(String.valueOf(messageCount))
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("PRODUCER_ACKS")
+                                .withValue("all")
+                            .endEnv()
+                            .addNewEnv()
+                                .withName("ADDITIONAL_CONFIG")
+                                .withValue(additionalConfig)
+                            .endEnv()
+                    .endContainer()
+                .endSpec()
+            .endTemplate()
+        .endSpec()
+        .build());
+```
+
+##### External using Vert.x client
+
+Lastly, external clients are implemented by supporting Vertx.io Verticles. This type of client has the most limitations for usage. Most notable is dependency on external listeners, which is caused by`platform dependency`. 
+They are implemented by _Abstract recursive builder, which makes these clients very handful. 
+What is worth mentioning is that external clients are running in same environment as your tests.
+Moreover, we have `Oauth`, `Tracing` and also `Basic` client. Each of them are for specific use case. 
+However, we prefer to always use the internal clients because of mentioned platform dependency. 
+Client can be used for instance:
+
+```
+// 1. Creating the client instance 
+
+BasicExternalKafkaClient basicExternalKafkaClient = new BasicExternalKafkaClient.Builder()
+    .withTopicName("topic-name")
+    .withNamespaceName("namespace-name")
+    .withClusterName("cluster-name")
+    .withKafkaUsername("kafka-username")
+    .withMessageCount(100)
+    .withSecurityProtocol(SecurityProtocol.SSL)
+    .build();
+
+// 2. Sending messages
+
+basicExternalKafkaClient.sendMesssageTls();
+```
+
+## <a id="dependencies">Dependencies</a>
+
+Before writing simple test case in Strimzi system tests, we need to know main dependencies that `system test` module has. 
+The most important ones are:
+1. test (kubernetes client)
+2. crd-generator + api (chaining / builder pattern)
```

Comment:
Do you really depend on crd-generator? I would probably just keep the `api` here.